File: tests/CCSharp.Demo/PrintAllInventoryContents.cs
using System;
using System.Linq;
using CCSharp.Attributes;
using CCSharp.ComputerCraft;
using CCSharp.ComputerCraft.Peripherals;

namespace CCSharp.Demo;

/// <summary>
/// This will print out all the items in every inventory connected to this computer
/// It will group up and sum stacks of the same item using linq
/// </summary>
[LuaProgram]
public static class PrintAllInventoryContents
{
    [LuaMain]
    public static void Start()
    {
        Inventory[] inventories = Peripheral.FindAll<Inventory>();
        foreach (Inventory inventory in inventories)
        {
            Console.WriteLine("Contents of: " + Peripheral.GetName(inventory));
            Console.WriteLine("------------------");
            foreach (var groupedItems in inventory.List().GroupBy(x => x.Name).ToArray())
            {
                string name = groupedItems.First().Name;
                Console.WriteLine(name + ": " + groupedItems.Sum(x => x.Count));
            }
            Console.WriteLine();
        }
    }
}

File: tests/CCSharp.Demo/lua/PrintAllInventoryContents.lua
local linq = require("linq")
function Start()
 local array = {peripheral.find("inventory")};
 local array2 = array;
 for _,inventory in ipairs(array2) do
  print("Contents of: "..tostring(peripheral.getName(inventory)))
  print("------------------")
  local array3 = linq.from(inventory.list()):groupBy(function(x)
  return x["name"];
end):toTable();
  for _,source in ipairs(array3) do
   local name = linq.from(source):first()["name"];
   print((name..": ")..tostring(linq.from(source):sum(function(x)
  return x["count"];
end)))
  end
  print()
 end
end
Start()